lib/src/file_util.rs

```diff
@@ -162,6 +165,26 @@ pub fn persist_content_addressed_temp_file<P: AsRef<Path>>(
     }
 }
 
+/// Reads from an async source and writes to a sync destination. Does not spawn
+/// a task, so writes will block.
+pub async fn copy_async_to_sync<R: AsyncRead, W: Write>(
+    mut reader: R,
+    mut writer: W,
+) -> io::Result<usize> {
```

Comment:
A similar function in tokio takes `&mut AsyncRead + Unpin` instead of pinning internally. I'm not sure which is better, and I don't think that would matter much (because traits are implemented for `&mut T`.)

https://docs.rs/tokio/1.45.0/tokio/io/fn.copy.html

For the writer side, taking `&mut W` might make more sense because the writer can be reused.

Reply:
I don't know which is better either. I updated the writer to be `&mut W`.